src/frontend/src/pages/Index/Settings/AdminCenter/TemplateManagementPanel.tsx

```diff
@@ -0,0 +1,261 @@
+import { t } from '@lingui/macro';
+import { Stack } from '@mantine/core';
+import { useMemo } from 'react';
+
+import { TemplatePreviewProps } from '../../../../components/editors/TemplateEditor/TemplateEditor';
+import { ApiFormFieldSet } from '../../../../components/forms/fields/ApiFormField';
+import { PanelGroup } from '../../../../components/nav/PanelGroup';
+import { ApiEndpoints } from '../../../../enums/ApiEndpoints';
+import { ModelType } from '../../../../enums/ModelType';
+import { InvenTreeIcon, InvenTreeIconType } from '../../../../functions/icons';
+import { TemplateTable } from '../../../../tables/settings/TemplateTable';
+
+type TemplateType = {
+  type: 'label' | 'report';
+  name: string;
+  singularName: string;
+  apiEndpoints: ApiEndpoints;
+  templateKey: string;
+  additionalFormFields?: ApiFormFieldSet;
+  defaultTemplate: string;
+  variants: {
+    name: string;
+    key: string;
+    icon: InvenTreeIconType;
+    preview: TemplatePreviewProps;
+  }[];
+};
+
+const defaultLabelTemplate = `{% extends "label/label_base.html" %}
```

Comment:
Can do that, but in the end after the refactor, it would make sense that each template type/variant can provide a default template via the api, so ther won't be any need to hardcode that in the frontend. Should I nevertheless move it into the defaults folder?

Reply:
Will make sense in the API but that still has to be sketched out. In general, I think it makes sense to have static, non-logical blocks split out. At some point, all of those should be served by APIs to make InvenTree as configurable as possible. That is much simpler if they are separated and organised.